.github/workflows/update_docs.yml

```
@@ -0,0 +1,76 @@
+name: Update Documentation
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build-docs:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # Fetch all history for doc
+
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.9'  # Replace '3.x' with your Python version
+
+      - name: Install dependencies
+        run: |
+          pip install -r doc/requirements.txt  # Install Sphinx and any other dependencies
+
+      - name: Build documentation
+        run: |
+          cd doc
+          make html
+          cd ..
+  
+      - name: Archive production artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: htmldoc
+          path: doc/_build/html/
+
+      - name: List directory contents Ffor debugging doc generation
```

Just a minor typo :)
```suggestion
      - name: List directory contents for debugging doc generation
```